session_rpcserver.go

```diff
@@ -315,15 +309,31 @@ func (s *sessionRpcServer) AddSession(_ context.Context,
 		}
 	}
 
+	// Reserve Session Key & ID.
+	id, localPrivKey, err := s.cfg.db.ReserveNewSessionID()
+	if err != nil {
+		return nil, err
+	}
+
+	cancelIDRes := func() {
+		err = s.cfg.db.CancelIDReservation(id)
+		if err != nil {
+			log.Errorf("could not cancel reservation for "+
+				"session ID %x", id)
+		}
+	}
+
 	sess, err := session.NewSession(
-		req.Label, typ, expiry, req.MailboxServerAddr, req.DevServer,
-		uniquePermissions, caveats, nil, false,
+		id, localPrivKey, req.Label, typ, expiry, req.MailboxServerAddr,
+		req.DevServer, uniquePermissions, caveats, nil, false, nil,
 	)
 	if err != nil {
+		cancelIDRes()
+
 		return nil, fmt.Errorf("error creating new session: %v", err)
 	}
 
-	if err := s.db.StoreSession(sess); err != nil {
+	if err := s.cfg.db.CreateSession(sess); err != nil {
 		return nil, fmt.Errorf("error storing session: %v", err)
```

Comment:
Should we also add `cancelIDRes()` here? As the `db.Update` that deletes the reservation in `CreateSession` won't be committed if `CreateSession` errors.

Reply:
indeed! good catch. I think im just doing to change it to a `defer` instead so that we never forget to call it if we exit before creating the session. If the session does get created successfully then the `defer cancelIDRes` will be a no-op :+1: